### PR TITLE
net: gptp: fix follow_up timestamp and announce byte order

### PR DESCRIPTION
**Pull request by chencaidy**
This PR include gPTP accuracy optimize and some byte order fix.

1. Change follow_up message preciseOriginTimestamp from software trigger to hardware egress capture
2. Fix announce message `cur_utc_offset` and `offset_scaled_log_var` fields byte order.

Using mimxrt1050-evk board and Intel i210 ethernet card direct cable connect, host using linuxptp stack,
and test PPS phase between 1588_EVENT2_OUT and SDP1 by oscilloscope.

Before:
mimxrt1050-evk as grandmaster, i210 as slave, PPS offset is 3.5ms
i210 as grandmaster, mimxrt1050-evk as slave, PPS offset is 20-30us

After:
mimxrt1050-evk as grandmaster, i210 as slave, PPS offset is 540-580ns
i210 as grandmaster, mimxrt1050-evk as slave, PPS offset is 1250-1380ns

### CODE REVIEW

File: subsys/net/l2/ethernet/gptp/gptp_messages.c [hunk 1]
```diff
@@ -267,14 +271,19 @@ struct net_pkt *gptp_prepare_follow_up(int port, struct net_pkt *sync)
 	hdr->ptp_version = GPTP_VERSION;
 	hdr->sequence_id = sync_hdr->sequence_id;
 	hdr->domain_number = 0U;
-	/* Store timestamp value in correction field. */
-	hdr->correction_field = gptp_timestamp_to_nsec(&sync->timestamp);
+	/* Gramd master clock should keep correction_field at zero */
```
Comment (jukkar) [inline subsys/net/l2/ethernet/gptp/gptp_messages.c:274]: Typo s/Gramd/Grand/

File: subsys/net/l2/ethernet/gptp/gptp_md.c [hunk 2]
```diff
@@ -38,36 +38,13 @@ static void gptp_md_follow_up_prepare(struct net_pkt *pkt,
 	hdr = GPTP_HDR(pkt);
```
Comment (jukkar) [inline subsys/net/l2/ethernet/gptp/gptp_md.c:38]: The `net: gptp: fix follow_up timestamp and announce byte order` contains three fixes. Could you separate them to three commits?
Please also enhance the commit messages so that they answer the question "Why this change is needed". The current changes look somewhat ok but it would be nice to have a bit more information (for example links to spec etc would be really helpful if possible).

Comment (chencaidy) [reply subsys/net/l2/ethernet/gptp/gptp_md.c:38]: Heys [jukkar](https://github.com/jukkar),
Already separated them to three commits, and add some reference in commit.
The document can be found at https://1.ieee802.org/wp-content/uploads/2019/03/802-1AS-rev-d8-0.pdf

**Review by jukkar — APPROVED** [review]
Thanks, LGTM

**Review by rlubos — APPROVED** [review]

**Comment by dleach02** [comment]
@chencaidy, thank you for the contribution! 

@hakehuang Can you test this PR to see how it does against #33747?

**Comment by chencaidy** [comment]
> @chencaidy, thank you for the contribution!
> 
> @hakehuang Can you test this PR to see how it does against #33747?

Here is my test report, may be able to offer some help.

OS: Ubuntu 22.04 with 5.19 kernel
Linuxptp version: [aa60db2](https://github.com/richardcochran/linuxptp/commit/aa60db270be12e7144e7b0cef2f5dde4213b7057)
testptp: https://elixir.bootlin.com/linux/v5.19/source/tools/testing/selftests/ptp/testptp.c

1. Build linuxptp and testptp (gcc -o testptp testptp.c)
2. Modity linuxptp/configs/gPTP.conf, `neighborPropDelayThresh 50000`, because mcux driver issue make path delay too high
3. Build gptp sample: west build -p always -b mimxrt1050_evk samples/net/gptp
4. If using mimxrt1050_evk as grandmaster, set `CONFIG_NET_GPTP_GM_CAPABLE=y` in prj.conf. using i120 as grandmaster, set to n, then flash to board.
5. Start host linuxptp: sudo ./ptp4l -f configs/gPTP.cfg -i `enp1s0` -m
6. Enable pps out in SDP1: sudo ./testptp -d /dev/ptp0 -L 1,2 -p 1000000000
If SDP1 pps out missing, rerun step 6.

Here is my test platform:
platform

- rt1052 as grandmaster before

linuxptp
linuxptp: path delay >18000ns, rms > 2000ns, sometimes have a jerk.

oscilloscope

- rt1052 as grandmaster after

linuxptp
linuxptp: delay < 900ns, rms < 50ns, no jerk.

oscilloscope

- host i210 as grandmaster before

oscilloscope

- host i210 as grandmaster after

oscilloscope

**Comment by hakehuang** [comment]
@chencaidy please take a looks at https://github.com/zephyrproject-rtos/zephyr/pull/60855 which should fix the jitter issue, and sync the timestamp in 24 hours
